src/java.base/share/classes/java/lang/reflect/Field.java

```diff
@@ -189,6 +183,10 @@ public void setAccessible(boolean flag) {
 
     @Override
     void checkCanSetAccessible(Class<?> caller) {
+        if (isTrustedFinal()) {
```

Comment:
I think our existing trusted final fields still can be set accessible; it means private fields values can be stolen, but the field themselves are still unmodifiable. So this logic would break existing behavior.

(For stable values, I think allowing reflection to steal the value is ok, except that such access does not respect memory order and may steal incomplete values)

Reply:
Ahh. This way is much better. I have provided such an implementation. Thanks for pointing this out.